File: src/main/java/io/github/nikpivkin/LifetimeService.java
package io.github.nikpivkin;

import java.util.Optional;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;

public interface LifetimeService {

  Lifetime initLifeTime(Player player);

  Lifetime increaseLifeTime(Player player, Lifetime lifetime);

  Lifetime decrementLifetime(Player player);

  Lifetime setLifetime(Player player, Lifetime lifetime);

  Lifetime getLifeTime(Player player);

  Lifetime takeTimeFromPlayer(Player player);

  Lifetime takeTimeFromPlayer(Player player, float percentage);

  Lifetime transferLifetimeToAnotherPlayer(Player from, Player to);

  Optional<Lifetime> getLifetimeAsReward(Player killer, Entity deadEntity);
}

File: src/main/kotlin/io/github/nikpivkin/lifetime/LifetimeService.kt
package io.github.nikpivkin.lifetime

import org.bukkit.entity.Entity
import org.bukkit.entity.Player
import java.util.*

interface LifetimeService {
    fun initLifeTime(player: Player): Lifetime

    fun increaseLifeTime(player: Player, lifetime: Lifetime): Lifetime

    fun decrementLifetime(player: Player): Lifetime

    fun setLifetime(player: Player, lifetime: Lifetime): Lifetime

    fun getLifeTime(player: Player): Lifetime

    fun takeTimeFromPlayer(player: Player): Lifetime

    fun takeTimeFromPlayer(player: Player, percentage: Float): Lifetime

    fun transferLifetimeToAnotherPlayer(from: Player, to: Player): Lifetime

    fun getLifetimeAsReward(killer: Player, deadEntity: Entity): Optional<Lifetime>
}
